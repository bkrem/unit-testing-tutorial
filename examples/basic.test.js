import React from 'react';
import {shallow} from 'enzyme';

const SimpleComponent = () => (<div>Hello world</div>);

describe('SimpleComponent', () => {

  // either use `it()`
  it('renders', () => {
    const renderedComponent = shallow(<SimpleComponent />);
    expect(renderedComponent).toMatchSnapshot();
  })

  // or `test()`
  test('component renders', () => {
    const renderedComponent = shallow(<SimpleComponent />);
    expect(renderedComponent).toMatchSnapshot();
  })

})
