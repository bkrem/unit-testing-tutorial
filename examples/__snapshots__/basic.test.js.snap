// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleComponent component renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SimpleComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Hello world",
    },
    "ref": null,
    "rendered": "Hello world",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello world",
      },
      "ref": null,
      "rendered": "Hello world",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SimpleComponent renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SimpleComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Hello world",
    },
    "ref": null,
    "rendered": "Hello world",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello world",
      },
      "ref": null,
      "rendered": "Hello world",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
